nwbinspector/checks/time_series.py

```diff
@@ -92,3 +92,38 @@ def check_resolution(time_series: TimeSeries):
         return InspectorMessage(
             message=f"'resolution' should use -1.0 or NaN for unknown instead of {time_series.resolution}."
         )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=TimeSeries)
+def check_rows_not_nan(time_series: TimeSeries, nelems=200):
+    """Check that each row of a TimeSeries has at least one non-NaN piece of data."""
+    n_dims = len(time_series.data.shape)
+    if n_dims > 2:
+        yield
+
+    subframe_selection = np.unique(
+        np.round(np.linspace(start=0, stop=time_series.data.shape[0] - 1, num=nelems)).astype(int)
+    )
```

Comment:
Looks like you could move this to line 110

Reply:
Using the clever logic we devised from the tables PR now (direct slicing with 'by' amount calculated from shape)